src/UnitTests/IMappingExpression/ForCtorParam.cs

```diff
@@ -51,5 +55,52 @@ public void Should_resolve_using_custom_func()
 
             dest.Value1.ShouldEqual(8);
         }
+
+        [Fact]
+        public void Should_ignore_nonexistent_parameter()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<Source, Dest>()
+                    .ForCtorParam("thing", opt => opt.MapFrom(src => src.Value))
+                    .ForCtorParam("think", opt => opt.MapFrom(src => src.Value));
```

Comment:
The point of this PR is exactly that it shouldn't matter, as long as there is still enough information for a valid mapping configuration.

Whether that behavior is desirable to more people than myself is of course debatable, but this is the behavior I want.